### PR TITLE
i18n(zh-cn): Add `error-reference.mdx` and 4 error page translations

### PR DESCRIPTION
**Pull request by XindiShang**
#### What kind of changes does this PR include?

- Translated content

#### Description

Added zh-cn `error-reference.mdx` translation and added:

- `no-matching-static-path-found.mdx`
- `static-client-address-not-available.mdx`
- `static-redirect-not-available.mdx`
-  `only-response-can-be-returned.mdx`

### CODE REVIEW

**Comment by liruifengv** [comment]
@XindiShang Thank you for the translation! Firstly, the documentation directory structure has been refactored， you can pull the latest code for the main branch and update it. 

Secondly, the English page adds a little `UnknowError` like https://github.com/withastro/docs/blob/main/src/content/docs/en/reference/error-reference.mdx?plain=1#L20.

You can find in https://github.com/withastro/docs/commit/efada0f8314479a4aef12184e7c2a8e683d431f9

File: src/pages/zh-cn/reference/errors/static-client-address-not-available.mdx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+i18nReady: true
+githubURL: https://github.com/withastro/astro/blob/main/packages/astro/src/core/errors/errors-data.ts
+---
+
+> **StaticClientAddressNotAvailable**: `Astro.clientAddress` 仅在设置 `output: 'server'` 时可用. 如果您需要服务端渲染，请更新您的 Astro 配置。(E03003)
+
+## 哪里出了问题？
+只有在启用[服务端渲染](/zh-cn/guides/server-side-rendering/)时，`Astro.clientAddress` 属性才可用。
+
+如要在静态模式下获取用户的IP地址，您可以在[客户端脚本](/zh-cn/core-concepts/astro-components/#客户端脚本) 中使用类似于 [Ipify](https://www.ipify.org/) 等不同的API，或者您可以使用在服务提供商所托管的无服务器 (serverless) 函数来获取用户的IP。
```
Comment (liruifengv) [inline src/pages/zh-cn/reference/errors/static-client-address-not-available.mdx:13]:
```suggestion
如要在静态模式下获取用户的IP地址，您可以在[客户端脚本](/zh-cn/core-concepts/astro-components/#客户端脚本) 中使用类似于 [Ipify](https://www.ipify.org/) 等不同的API，或者您可以使用托管在服务提供商处的无服务器 (serverless) 函数来获取用户的IP。
```

**Review by liruifengv — CHANGES_REQUESTED** [review]

File: src/pages/zh-cn/reference/errors/static-redirect-not-available.mdx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+i18nReady: true
+githubURL: https://github.com/withastro/astro/blob/main/packages/astro/src/core/errors/errors-data.ts
+---
+
+> **StaticRedirectNotAvailable**: 重定向仅在设置 `output: 'server'` 时可用. 如果您需要服务端渲染，请更新您的 Astro 配置。(E03001)
+
+## 哪里出了问题？
+只有在启用[服务端渲染](/zh-cn/guides/server-side-rendering/)时，`Astro.redirect` 函数才可用。
+
+如要在静态网站上重定向，可以使用 [meta 标签的 refresh 属性](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta)。某些站点托管平台也会提供基于配置的重定向（例如：[Netlify 重定向](https://docs.netlify.com/routing/redirects/)）。
```
Comment (liruifengv) [inline src/pages/zh-cn/reference/errors/static-redirect-not-available.mdx:13]:
```suggestion
如果要在静态网站上重定向，可以使用 [meta 标签的 refresh 属性](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta)。某些站点托管平台也会提供基于配置的重定向（例如：[Netlify 重定向](https://docs.netlify.com/routing/redirects/)）。
```

File: src/pages/zh-cn/reference/errors/no-matching-static-path-found.mdx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+---
+layout: ~/layouts/MainLayout.astro
+title: No static path found for requested path.
+i18nReady: true
+githubURL: https://github.com/withastro/astro/blob/main/packages/astro/src/core/errors/errors-data.ts
+---
+
+> **NoMatchingStaticPathFound**: 匹配了 `getStaticPaths()` 路由模式，但未找到与请求路径 `PATH_NAME` 所匹配的静态路径。 (E03004)
```
Comment (liruifengv) [inline src/pages/zh-cn/reference/errors/no-matching-static-path-found.mdx:8]:
```suggestion
> **NoMatchingStaticPathFound**: 一个 `getStaticPaths()` 中的路由模式被匹配到了，但未找到与请求路径 `PATH_NAME` 所匹配的静态路径。 (E03004)
```

File: src/pages/zh-cn/reference/errors/static-client-address-not-available.mdx (hunk 1 above)
Comment (XindiShang) [reply src/pages/zh-cn/reference/errors/static-client-address-not-available.mdx:13]: Thanks for the review! I'll focus more on readability.

**Review by liruifengv — APPROVED** [review]
LGTM! Thank you for your continued follow-up. @XindiShang 

@delucis I reviewed this PR. Please do a final check. Thanks.

**Review by Yan-Thomas — COMMENTED** [review]
Thanks for the hard work @XindiShang & @liruifengv, I added some suggestions regarding MDN links available in zh-CN, after this LGTM.

File: src/content/docs/zh-cn/reference/errors/only-response-can-be-returned.mdx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+---
+title: Invalid type returned by Astro page.
+i18nReady: true
+githubURL: https://github.com/withastro/astro/blob/main/packages/astro/src/core/errors/errors-data.ts
+---
+
+> 路由页面返回了一个错误的返回值 `RETURNED_VALUE`。在 Astro 文件中只能返回 `Response`。(E03005)
+
+## 哪里出了问题？
+在 Astro 文件中只能返回 [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) 的实例。
```
[the remark this answers is not in the document]
Comment (XindiShang) [reply src/content/docs/zh-cn/reference/errors/only-response-can-be-returned.mdx]: Very thoughtful suggestion! Thanks for the review.

**Pull request merged**